sources/light_client.move

```diff
@@ -113,17 +121,24 @@ public entry fun insert_headers(lc: &mut LightClient, raw_headers: vector<vector
     assert!(!raw_headers.is_empty(), EHeaderListIsEmpty);
 
     let first_header = new_block_header(raw_headers[0]);
-    let latest_block = lc.latest_block();
+    let head = lc.head();
 
     let mut is_forked = false;
-    if (first_header.prev_block() == latest_block.header().block_hash()) {
+    if (first_header.prev_block() == head.header().block_hash()) {
         // extend current fork
         lc.extend_chain(first_header.prev_block(), raw_headers);
     } else {
         // handle a fork choice
-        assert!(lc.exist(first_header.prev_block()), EBlockNotFound);
-        let current_chain_work = latest_block.chain_work();
-        let current_block_hash = latest_block.header().block_hash();
+        let parent_id = first_header.prev_block();
+        assert!(lc.exist(parent_id), EBlockNotFound);
+        let parent = lc.get_light_block_by_hash(parent_id);
+        // TODO: check if we need to block extending beyond finality
+        // * pro: we protect against double mint for nBTC etc...
+        // * cons: we can have a deadlock... probably we should remove
+        assert!(lc.head_height - parent.height() <= FINALITY, EExtendBeyondFinal);
```

Comment:
Add test for case EExtendBeyondFinal.

Reply:
I was thinking about this - deadlock is more dangerous, s we should not have that assert, I will keep it as a comment.

Reply:
I've added an event instead of the error.